vm/src/eval.rs

```diff
@@ -277,17 +286,65 @@ pub fn eval_inst(vm: &mut NexusVM<impl Memory>) -> Result<()> {
         ECALL { rd } => {
             RD = rd;
             vm.Z = vm.syscalls.syscall(vm.regs.pc, vm.regs.x, &vm.mem)?;
+
+            // Collects label from syscall and starts counting cycles
+            if vm.Z == (SyscallCode::ProfileCycles as u32) {
+                if let Some(s) = vm.syscalls.get_label() {
+                    let label = std::str::from_utf8(&s).unwrap();
+                    let fn_name = label.split(':').last().unwrap().to_string();
```

Comment:
`from_utf8` is indeed unlikely to panic. The second line, however, should be handled.

Reply:
Added `expect` to show error when the assumption `:` does not hold.